### PR TITLE
Split the images directory even more

### PR DESCRIPTION
**Pull request by coolo**
As we have too many images, splitting them in 256 directories is not good enough.
On a plus we avoid /image/ad/*

No migration yet, the old symlinks still work - and will not output /image/ad/* but
/tests/X/image, which makes us loose the advantage that the browser knows it already
has seen that image. But that's a task for later

### CODE REVIEW

**Comment by coolo** [comment]
@okurz not really an idea how to test this properly - the tests passed even when all thumbnails pointed to invalid URLs

File: lib/OpenQA/Schema/Result/JobModules.pm [hunk 1]
```diff
@@ -119,8 +119,13 @@ sub details() {
...
         next unless $link;
-        $img->{md5_basename} = basename($link);
-        $img->{md5_dirname}  = basename(dirname($link));
+        my $base = basename($link);
+        my $dir  = dirname($link);
+        $dir =~ s,^.*/images/,,;
+        if ($dir ne 'ad') {    # avoid problem with ad block
+            $img->{md5_dirname}  = $dir;
+            $img->{md5_basename} = $base;
+        }
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/JobModules.pm]: this is like a very specific super special workaround which will bite us in the ass eventually, I assume. I'd rather ask every user to disable adblock on openqa hosts until we have a proper avoidance using migration of current results.

**Comment by okurz** [comment]
first commit is ok-ish (I guess it will break something), the second I would avoid for now. Maybe we can think of something better over the next days, I don't think the issue is urgent, considering we had this issue already for some years, I guess.

File: lib/OpenQA/Schema/Result/JobModules.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Schema/Result/JobModules.pm]: The 'ad' directory will go away over time anyway - but I think that if is an acceptable workaround for confusions like yours. Migrating the images is another solution, but I will not do that soonish as it's a pain to test.

**Comment by coolo** [comment]
I removed ad blocker blocker again

**Comment by aaannz** [comment]
If the links to images will now be /tests/X/image, wouldn't this put another strain to mojo?
When images are distributed under /images it is done by apache directly.

I'm for to rather inform users to whitelist openqa in adblockers and eventually we either do migration or the results will simply expire.

**Comment by coolo** [comment]
the links for new tests will be as before - just one level deeper. I'd love to migrate old images, but it's a lot of work and will take forever

**Comment by coolo** [comment]
(which is why I asked you weeks ago to do that for me ;)

**Comment by okurz** [comment]
LGTM

**Comment by coolo** [comment]
I meanwhile think I want to put this in production together with a migration, so I leave it idling till I get around to add it.

**Comment by okurz** [comment]
now?

**Comment by sysrich** [comment]
now?

**Comment by coolo** [comment]
no

**Comment by dirkmueller** [comment]
now?

**Comment by coolo** [comment]
is this some running gag?

**Comment by okurz** [comment]
No, it's not. It's all the peoples loosing time until the come to the "aaah, adblocker!" conclusion :-) see discussion at 13/09/2016 15:43:48 (UTC or CEST, I don't know) and before in #opensuse-factory.

**Comment by coolo** [comment]
I created the migration - for easier testing hard coded into gru startup. But I'm afraid I will have to create a schema change (I'm starting to hate dbix for the way it handles migrations).

So for review sake assume the gru jobs are scheduled in a migration. But there are 3 sets of GRU tasks run after each other by means of different priority levels (and the knowledge that we only have one gru):

- migrate images/XX/REST.png to images/XXX/YYY/REST.png creating compat symlink
- going through testresults/ID and redoing the symlinks from the XX/REST to the XXX/YYY/REST.png format
- removing all images/XX

**Comment by coolo** [comment]
the webui will be able to show proper thumbnails all the time - and it will take a good part of the month till this is done :)

**Comment by coolo** [comment]
-0.6% was a bit heavy - so let's see what my test cases bring

**Comment by okurz** [comment]
you did good :-)

**Comment by coolo** [comment]
with this I'll merge now :)

**Comment by coolo** [comment]
this is now deployed on openqa.opensuse.org - and it shows that XXX/YYY is a bit over the top. Most if not all of these directories contain a single file now :)

But I don't think it harms.

**Comment by okurz** [comment]
future proof then :-)

**Pull request merged**